src/temp_sensor.rs

```diff
@@ -0,0 +1,165 @@
+            _ => Self::Default,
+        }
+    }
+}
+
+// -- TemperatureSensorConfig --
+
+#[cfg(not(any(esp32)))]
+#[derive(Copy, Clone)]
+/// Rust wrapper for `temperature_sensor_config_t`
+pub struct TemperatureSensorConfig {
+    // TODO: check int size
+    pub range_min: i32,
+    pub range_max: i32,
+    pub clk_src: TemperatureSensorClockSource,
+}
+
+#[cfg(not(any(esp32)))]
+impl From<temperature_sensor_config_t> for TemperatureSensorConfig {
+    fn from(value: temperature_sensor_config_t) -> Self {
+        TemperatureSensorConfig {
+            range_min: value.range_min,
+            range_max: value.range_max,
+            clk_src: value.clk_src.into(),
+        }
+    }
+}
+
+#[cfg(not(any(esp32)))]
+impl Into<temperature_sensor_config_t> for TemperatureSensorConfig {
+    fn into(self) -> temperature_sensor_config_t {
+        temperature_sensor_config_t {
+            clk_src: self.clk_src.into(),
+            range_max: self.range_max,
+            range_min: self.range_min,
+        }
+    }
+}
+
+#[cfg(not(any(esp32)))]
+impl Default for TemperatureSensorConfig {
+    fn default() -> Self {
+        TemperatureSensorConfig {
+            range_min: -10,
+            range_max: 80,
+            clk_src: TemperatureSensorClockSource::Default,
+        }
+    }
+}
+
+// -- TemperatureSensorDriver --
+
+#[cfg(not(any(esp32)))]
+pub struct TemperatureSensorDriver {
+    ptr: temperature_sensor_handle_t,
+    // To track current state (to avoid double enable/disable)
+    enabled: bool,
+}
+
+#[cfg(not(any(esp32)))]
+impl TemperatureSensorDriver {
+    pub fn new(config: TemperatureSensorConfig) -> Self {
+        let mut sensor = core::ptr::null_mut();
+        unsafe {
+            temperature_sensor_install(&config.into(), &mut sensor);
+        }
+        TemperatureSensorDriver {
+            ptr: sensor,
+            enabled: false,
+        }
+    }
+
+    pub fn enable(&mut self) {
+        // TODO: Perhaps handing the "double enable" error to the user makes more sense
+        if self.enabled {
+            return;
+        }
+        self.enabled = true;
+        unsafe { temperature_sensor_enable(self.ptr) };
+    }
+
+    pub fn disable(&mut self) {
+        // TODO: Perhaps handing the "double disable" error to the user makes more sense
+        if !self.enabled {
+            return;
+        }
+        self.enabled = false;
+        unsafe { temperature_sensor_disable(self.ptr) };
+    }
+
+    pub fn get_celsius(&self) -> f32 {
+        let mut val = 0.0;
+        unsafe { temperature_sensor_get_celsius(self.ptr, &mut val) };
+        val
+    }
+
+    pub fn get_fahrenheit(&self) -> f32 {
+        let celsius = self.get_celsius();
+        (celsius * 1.8) + 32.0
+    }
+
+    pub fn get_kelvin(&self) -> f32 {
+        self.get_celsius() + 273.15
+    }
+}
+
+#[cfg(not(any(esp32)))]
+impl Drop for TemperatureSensorDriver {
+    fn drop(&mut self) {
+        unsafe {
+            self.disable();
```

Comment:
self.disable is not an unsafe call - try to scope unsafe to the minimal part where it is originated from

Reply:
Done in 2b0694d067e75c1ffbe118c3a890e6b6801fb5c8